### ISSUE TITLE
Cannot read property 'previous' of null when running polymer build

### ISSUE THREAD

**Issue by Davis22xd** [Polymer/polymer-cli#305]
### Description

When I was trying to build my polymer project with polymer build I got the error:
cannot read property 'previous' of null
### Versions & Environment

- Polymer CLI: 0.11.1
- node: v4.2.1
- Operating System: Ubuntu 14.04

#### Steps to Reproduce

1. I create my project that I have in:
**https://github.com/Davis22xd/polymer-table**
2. checkout to origin/simple-table branch
3. I try to build the project with polymer build

#### Expected Results

vulcanized file 

#### Actual Results

info:    Building application...
warn:    Unable to uglify file /home/sumitrag-david/Documents/Proyectos/polymer-table/polymer-table.html_script_0.js
info:    Generating build/unbundled...
info:    Generating build/bundled...
error:   Uncaught exception: TypeError: Cannot read property 'previous' of null
error:   TypeError: Cannot read property 'previous' of null
    at Parser.parseDeclarationOrRuleset (/usr/local/lib/node_modules/polymer-cli/node_modules/shady-css-parser/dist/shady-css/parser.js:192:63)
    at Parser.parseRule (/usr/local/lib/node_modules/polymer-cli/node_modules/shady-css-parser/dist/shady-css/parser.js:80:21)
    at Parser.parseRules (/usr/local/lib/node_modules/polymer-cli/node_modules/shady-css-parser/dist/shady-css/parser.js:62:25)
    at Parser.parseStylesheet (/usr/local/lib/node_modules/polymer-cli/node_modules/shady-css-parser/dist/shady-css/parser.js:55:47)
    at Parser.parse (/usr/local/lib/node_modules/polymer-cli/node_modules/shady-css-parser/dist/shady-css/parser.js:51:19)
    at css (/usr/local/lib/node_modules/polymer-cli/node_modules/css-slam/index.js:51:39)
    at GulpTransform._transform (/usr/local/lib/node_modules/polymer-cli/node_modules/css-slam/index.js:69:36)
    at GulpTransform.Transform._read (_stream_transform.js:167:10)
    at GulpTransform.Transform._write (_stream_transform.js:155:12)
    at doWrite (_stream_writable.js:292:12)

**Comment by oneezy** [comment]
Any luck on this @Davis22xd ?

I created duplicate around exact same time you did.. issue #306

**Comment by Davis22xd** [comment]
No yet, but I'm looking for it.  I hope to find something

**Comment by oneezy** [comment]
Using `polymer build --verbose`,

I'm getting these syntax errors:

```
'SyntaxError: Unexpected character \'#\'',
'SyntaxError: Unexpected character \'`\''
```

**Source Code:** https://github.com/oneezy/shopSFI

**Full output:**
```
Oneeezy@Oneezy MINGW64 ~/Desktop/www_/_clients/diesel/wwwSFI (firebase)
$ polymer build --verbose
debug:   got args:
{ args: [ 'build', '--verbose' ] }
debug:   got default config from file:
{ config:
   { entrypoint: 'index.html',
     shell: 'src/shop-app.html',
     fragments:
      [ 'src/shop-list.html',
        'src/shop-detail.html',
        'src/shop-cart.html',
        'src/shop-checkout.html',
        'src/app-resources.html' ] } }
debug:   adding command build
debug:   adding command help
debug:   adding command init
debug:   adding command lint
debug:   adding command serve
debug:   adding command test
debug:   running...
debug:   command 'build' found, parsing command args:
{ args: [ '--verbose' ] }
debug:   command options parsed from args:
{ 'sw-precache-config': 'sw-precache-config.js', verbose: true }
debug:   final project configuration generated:
{ root: 'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI',
  entrypoint: 'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\index.html',
  shell: 'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-app.html',
  fragments:
   [ 'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-list.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-detail.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-cart.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-checkout.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\app-resources.html' ],
  inputs:
   [ 'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\index.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-app.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-list.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-detail.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-cart.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\shop-checkout.html',
     'C:\\Users\\Oneeezy\\Desktop\\www_\\_clients\\diesel\\wwwSFI\\src\\app-resources.html' ] }
debug:   Running command...
debug:   building with options
{ sources: undefined,
  swPrecacheConfig: 'sw-precache-config.js',
  insertDependencyLinks: undefined,
  html: {},
  css: {},
  js: {} }
debug:   building via standard build()...
debug:   shell: C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-app.html
debug:   entrypoint: C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\index.html
debug:   dependencies: C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\**\*
debug:   sources: C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\index.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-app.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-list.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-detail.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-cart.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-checkout.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\app-resources.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\**\*,!build,!build/**/*,!bower_components,!bower_components/**/*,!node_modules,!node_modules/**/*
debug:   fragments: C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-list.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-detail.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-cart.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\shop-checkout.html,C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\src\app-resources.html
debug:   no gulpfile found (searched up from C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI)
info:    Building application...
debug:   Reading source files...
debug:   Reading dependencies...
debug:   Analyzing build dependencies...
info:    Generating build/unbundled...
info:    Generating build/bundled...
error:   Uncaught exception: TypeError: Cannot read property 'previous' of null
error:   TypeError: Cannot read property 'previous' of null
    at Parser.parseDeclarationOrRuleset (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:192:63)
    at Parser.parseRule (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:80:21)
    at Parser.parseRulelist (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:153:27)
    at Parser.parseDeclarationOrRuleset (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:212:87)
    at Parser.parseRule (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:80:21)
    at Parser.parseRules (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:62:25)
    at Parser.parseStylesheet (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:55:47)
    at Parser.parse (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\shady-css-parser\dist\shady-css\parser.js:51:19)
    at css (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\css-slam\index.js:51:39)
    at dom5.queryAll.forEach.styleNode (C:\Users\Oneeezy\AppData\Roaming\nvm\v6.3.0\node_modules\polymer-cli\node_modules\css-slam\index.js:45:36)
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\marked-element\demo\index.html_script_0.js
debug:
{ message: 'SyntaxError: Unexpected character \'#\'',
  filename: 0,
  line: 2,
  col: 10,
  pos: 11,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at parse_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1677:9)\n    at Object.next_token [as input] (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1946:9)\n    at next (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2075:25)\n    at Object.parse (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2061:15)\n    at addFile (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:69:33)\n    at C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:77:17\n    at Array.forEach (native)\n    at Object.exports.minify (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:75:26)' }
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\marked-element\test\marked-element.html_script_0.js
debug:
{ message: 'SyntaxError: Unexpected character \'#\'',
  filename: 0,
  line: 2,
  col: 8,
  pos: 9,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at parse_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1677:9)\n    at Object.next_token [as input] (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1946:9)\n    at next (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2075:25)\n    at Object.parse (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2061:15)\n    at addFile (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:69:33)\n    at C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:77:17\n    at Array.forEach (native)\n    at Object.exports.minify (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:75:26)' }
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\marked-element\test\marked-element.html_script_1.js
debug:
{ message: 'SyntaxError: Unexpected character \'`\'',
  filename: 0,
  line: 2,
  col: 0,
  pos: 1,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at parse_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1677:9)\n    at Object.next_token [as input] (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1946:9)\n    at next (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2075:25)\n    at Object.parse (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2061:15)\n    at addFile (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:69:33)\n    at C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:77:17\n    at Array.forEach (native)\n    at Object.exports.minify (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:75:26)' }
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\marked-element\test\marked-element.html_script_2.js
debug:
{ message: 'SyntaxError: Unexpected character \'`\'',
  filename: 0,
  line: 2,
  col: 0,
  pos: 1,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at parse_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1677:9)\n    at Object.next_token [as input] (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1946:9)\n    at next (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2075:25)\n    at Object.parse (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2061:15)\n    at addFile (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:69:33)\n    at C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:77:17\n    at Array.forEach (native)\n    at Object.exports.minify (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:75:26)' }
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\marked-element\test\marked-element.html_script_3.js
debug:
{ message: 'SyntaxError: Unexpected character \'`\'',
  filename: 0,
  line: 2,
  col: 0,
  pos: 1,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at parse_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1677:9)\n    at Object.next_token [as input] (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1946:9)\n    at next (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2075:25)\n    at Object.parse (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2061:15)\n    at addFile (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:69:33)\n    at C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:77:17\n    at Array.forEach (native)\n    at Object.exports.minify (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:75:26)' }
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\marked-element\test\marked-element.html_script_4.js
debug:
{ message: 'SyntaxError: Unexpected character \'`\'',
  filename: 0,
  line: 2,
  col: 0,
  pos: 1,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at parse_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1677:9)\n    at Object.next_token [as input] (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1946:9)\n    at next (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2075:25)\n    at Object.parse (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2061:15)\n    at addFile (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:69:33)\n    at C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:77:17\n    at Array.forEach (native)\n    at Object.exports.minify (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:75:26)' }
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\web-component-tester\data\index.html_script_0.js
debug:
{ message: 'SyntaxError: Unexpected token: operator (<)',
  filename: 0,
  line: 1,
  col: 6,
  pos: 6,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at croak (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2089:9)\n    at token_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2097:9)\n    at unexpected (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2103:9)\n    at expr_atom (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2630:9)\n    at maybe_unary (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2792:19)\n    at expr_ops (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2827:24)\n    at maybe_conditional (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2832:20)\n    at maybe_assign (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2856:20)' }
warn:    Unable to uglify file C:\Users\Oneeezy\Desktop\www_\_clients\diesel\wwwSFI\bower_components\web-component-tester\data\index.html_script_2.js
debug:
{ message: 'SyntaxError: Unexpected token: operator (<)',
  filename: 0,
  line: 2,
  col: 21,
  pos: 22,
  stack: 'Error\n    at new JS_Parse_Error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1545:18)\n    at js_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:1553:11)\n    at croak (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2089:9)\n    at token_error (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2097:9)\n    at unexpected (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2103:9)\n    at expr_atom (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2630:9)\n    at maybe_unary (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2792:19)\n    at expr_ops (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2827:24)\n    at maybe_conditional (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2832:20)\n    at maybe_assign (eval at <anonymous> (C:\\Users\\Oneeezy\\AppData\\Roaming\\nvm\\v6.3.0\\node_modules\\polymer-cli\\node_modules\\uglify-js\\tools\\node.js:1:0), <anonymous>:2856:20)' }

```

**Comment by oneezy** [comment]
It looks as if this might be a css syntax problem (shady css parser ?).

@justinfagnani, @leesander1 any update on this or best way to handle it? 

**raydaly from SO wrote:**
[(source)](https://stackoverflow.com/questions/37473722/polymer-build-error)

> I was seeing the same error message of "Cannot read property 'previous' of null" when using polymer build for six files. I found an error in the CSS for one of my component files. Specifically, instead of curly brackets I had used regular brackets. Once corrected, everything built correctly.

> So I suggest that you examine the CSS for each component file. Probably a CSS error is stopping the build. To find the error, I went file by file and deleted the CSS until the build would work correctly. Once I had the component file that contained the bad CSS, I then deleted one line at a time until I found the error. If you CSS is long, you might try CSS lint.


**Alan from SO wrote:**
[(source)](http://stackoverflow.com/questions/37449621/missing-component-html-script-0-js-files-in-polymer-build-using-the-cli-tool/37471089#37471089)
> I had this same problem and saw that there were several issues #191 reporting it.

> It seems a fix has been already merged #199 and is part of [v0.9.0](https://github.com/Polymer/polymer-cli/compare/v0.8.0...master) but I'm not completely sure if the fix is up on npm just yet.

**Comment by FredKSchott** [comment]
I can reproduce this as well. It seems like CSS errors will stop the build stream, so make sure you don't have any in your source files. I'll be doing some work around this soon to turn these into warnings (similar to how uglify warns but doesn't exit).

**Comment by oneezy** [comment]
@FredKSchott , I ended up fixing this by deleting some unused folder/files that I had pulled in to reference from an older version of my project..but I'm unsure if it was tied directly to making the build fail. 

If files exist within your app that aren't being used, do they still get processed into the build to cause these errors? It would be nice if the CLI automatically knew to not include these..or maybe adding an extra command `polymer build minimal` or something.

Also, I'm curious why `polymer build` minifies files, but **does not** concatenate files together? Is this the correct behavior?

`build > bundled > bower_components`:
![image](https://cloud.githubusercontent.com/assets/7369575/17200358/085069d0-544a-11e6-89fa-07952d4f7595.png)

**Comment by FredKSchott** [comment]
@oneezy the latest published version of polymer-cli (naively) includes all files that could possibly be needed, which includes your bower_components directory by default (:-1:). The next version (soon to be published) uses our new [polymer-build](https://github.com/Polymer/polymer-build) library, which is now smart enough to know which files your build needs and which it does not. 

Feel free to check out the master branch and use the new build pipeline today. And we're getting ready to publish a new release in the next few days.

**Comment by oneezy** [comment]
Great news! Thanks for the update. I'll check it out.

**Comment by FredKSchott** [comment]
Fixed in [v0.13.0](https://github.com/Polymer/polymer-cli/releases/tag/v0.13.0)

**Comment by Davis22xd** [comment]
Great new thanks
